agent/module_agent.py

```diff
@@ -0,0 +1,78 @@
+from agent.base_agent import Agent
+import math
+import numpy as np
+
+class ModuleAgent(Agent):
+    def __init__(self, module, ogm, max_steps=1000):
+        super().__init__()
+        self.max_steps = max_steps
+        self.steps_taken = 0
+        self.success = False
+        self.module = module
+        self.ogm = ogm
+        self.calc_final_coords()
+        self.calc_final_local_maps()
+        self.past_v_final_local_map = set() 
+        self.curr_v_final_local_map = self.calc_curr_v_final_local_map()
+        self.past_v_final_local_map = self.curr_v_final_local_map
+        
+        
+        
+    def get_coords(self, ogm):
+        return ogm.module_positions[self.module]
+    
+    def calc_final_coords(self):
+        self.final_coords = []
+        for i in range(len(self.ogm.final_rotated_module_positions)):
+            temp_pos = self.ogm.final_rotated_module_positions[i]
+            self.final_coords.append(temp_pos[self.module])
+    
+    def get_local_map(self):
+        origin = self.get_coords(self.ogm)
+        ogm_shape = self.ogm.curr_grid_map.shape
+        limits = [origin[0] - 2, origin[0] + 2, origin[1] - 2, origin[1] + 2, origin[2] - 2, origin[2] + 2]
+
+        for i in range(6):
+            if limits[i] < 0:
+                limits[i] =  0
+            elif limits[i] >= ogm_shape[math.floor(i / 2)]:
+                limits[i] = ogm_shape[math.floor(i / 2)] - 1
+        
+        return self.ogm.curr_grid_map[limits[0]:(limits[1] + 1), limits[2]:(limits[3] + 1), limits[4]:(limits[5] + 1)]
+    
+    def calc_final_local_maps(self):
+        self.final_local_maps = []
+
+        for i in range(len(self.ogm.final_grid_maps)):
+            origin = self.final_coords[i]
+            ogm_shape = self.ogm.curr_grid_map.shape
+            limits = [origin[0] - 2, origin[0] + 2, origin[1] - 2, origin[1] + 2, origin[2] - 2, origin[2] + 2]
+            temp_final_map = self.ogm.final_grid_maps[i]
+
+            for i in range(6):
+                if limits[i] < 0:
+                    limits[i] =  0
+                elif limits[i] >= ogm_shape[math.floor(i / 2)]:
+                    limits[i] = ogm_shape[math.floor(i / 2)] - 1
+            self.final_local_maps.append(temp_final_map[limits[0]:limits[1] + 1, limits[2]:limits[3] + 1, limits[4]:limits[5] + 1])
+            
+    def calc_curr_v_final_local_map(self):
```

Comment:
Is it ok to have the module 1 compared to 5th rotation local map while module 2 compared 2 2nd rotation local map ... haven't thought through this completely but among the two cases, 

each module will be rewarded if it matches with any of the two? is it good reward, or we should only be rewarding with the rotation that's giving the most matching in all the modules?